imports/ui/components/editor/components/streamEditorComponent.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'react-komposer';
+
+import { Row, Column, Flex } from 'glamor/jsxstyle';
+import Snackbar from 'material-ui/Snackbar';
+
+import trackerLoader from '../../../../util/tracker-loader';
+import { StreamEditorContent } from '../../../../collections';
+import { setStreamEditorContent, initStreamEditorModel } from '../../../../redux/modules/editor';
+
+import { makeEditorModel } from '../../../../util/editor-model-factory';
+import CodemoEditor from './codemoEditor';
+
+class StreamEditorComponent extends CodemoEditor {
+  constructor() {
+    super();
+    this.container = 'stream_monaco_container';
+    this.LEADER_MESSAGE_DURATION = 2500;
+    this.state = {
+      showLeaderMessage: false,
+    };
+  }
+
+  monacoDidInit() {
+    this.props.dispatch(initStreamEditorModel(window.monaco.editor));
```

Comment:
Here we should set up the event listener for cursor change, but just do nothing if not leading (otherwise we'll hit infinite update loops).

```js
this.editor.onDidChangeCursorPosition(() => {
  if (NOT_THE_LEADER) return;

  // Update the redux state w/
  // this.editor.getValue()
  // this.editor.saveViewState()
});
```